test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/services/chat-states/ChatStates.spec.js

```diff
@@ -321,6 +377,64 @@ describe("Chat States", () => {
             expect(container.transcript.addSystemMsg).toHaveBeenCalledWith({msg: "You're now talking to Jay"});
         });
 
+        it("sends AgentTyping to the transcript", () => {
+            const [sdk, container] = createEngagedStateDependencies();
+
+            const state = new ChatStates.EngagedState(sdk, container, [], jest.fn());
+
+            let agentIsTypingSpy = jest.spyOn(state, '_chatActivityAndAgentTyping');
+
+            const handleMessage = sdk.getMessages.mock.calls[0][0];
+            const message = {
+                data: {
+                    "type": "2",
+                    "state": "agentIsTyping",
+                    "agentID": "42409619",
+                    "user.id":"42409619",
+                    "sessionId": "2493130538282329498",
+                    "aeapi.mode": "true",
+                    "messageType":"chat.activity",
+                    "display.text": "Agent is typing...",
+                    "engagementID": "388260662637696059",
+                    "external_user.ip": "80.0.102.28",
+                    "config.session_id": "2493130538282329498",
+                    "msg.originalrequest.id": "2493130538484377173"
+                }
+            };
+
+            handleMessage(message);
+            expect(agentIsTypingSpy).toBeCalledTimes(1);
```

Comment:
this test should also expect the transcript to be called with the expected object, potentially you could use .toMatchObject for the assertion